data/scripts/lib/register_monster_type.lua

```diff
@@ -394,120 +277,119 @@ registerMonsterType.defenses = function(mtype, mask)
-								spell:setConditionDamage(defense.condition.minDamage, defense.condition.maxDamage, startDamage)
-							end
-							if defense.condition.duration then
-								spell:setConditionDuration(defense.condition.duration)
-							end
-							if defense.condition.interval then
-								spell:setConditionTickInterval(defense.condition.interval)
-							end
-						end
-					else
-						spell:setType(defense.name)
-						if defense.type then
-							if defense.name == "combat" then
-								spell:setCombatType(defense.type)
-							else
-								spell:setConditionType(defense.type)
-							end
-						end
-						if defense.interval then
-							spell:setInterval(defense.interval)
-						end
-						if defense.chance then
-							spell:setChance(defense.chance)
-						end
-						if defense.range then
-							spell:setRange(defense.range)
-						end
-						if defense.duration then
-							spell:setConditionDuration(defense.duration)
-						end
-						if defense.speed then
-							if type(defense.speed) ~= "table" then
-								spell:setConditionSpeedChange(defense.speed)
-							elseif type(defense.speed) == "table" then
-								if defense.speed.min and defense.speed.max then
-									spell:setConditionSpeedChange(defense.speed.min, defense.speed.max)
-								end
-							end
-						end
-						if defense.target then
-							spell:setNeedTarget(defense.target)
-						end
-						if defense.length then
-							spell:setCombatLength(defense.length)
-						end
-						if defense.spread then
-							spell:setCombatSpread(defense.spread)
-						end
-						if defense.radius then
-							spell:setCombatRadius(defense.radius)
-						end
-						if defense.minDamage and defense.maxDamage then
-							if defense.name == "combat" then
-								spell:setCombatValue(defense.minDamage, defense.maxDamage)
-							else
-								local startDamage = 0
-								if defense.startDamage then
-									startDamage = defense.startDamage
-								end
-								spell:setConditionDamage(defense.minDamage, defense.maxDamage, startDamage)
-							end
-						end
-						if defense.effect then
-							spell:setCombatEffect(defense.effect)
-						end
-						if defense.shootEffect then
-							spell:setCombatShootEffect(defense.shootEffect)
-						end
-					end
-				elseif defense.script then
-					spell:setScriptName(defense.script)
-					if defense.interval then
-						spell:setInterval(defense.interval)
-					end
-					if defense.chance then
-						spell:setChance(defense.chance)
-					end
-					if defense.minDamage and defense.maxDamage then
-						spell:setCombatValue(defense.minDamage, defense.maxDamage)
-					end
-					if defense.target then
-						spell:setNeedTarget(defense.target)
-					end
-					if defense.direction then
-						spell:setNeedDirection(defense.direction)
+				local spell = AbilityTableToSpell(defense)
+				mtype:addDefense(spell)
+			end
+		end
+	end
+end
+function AbilityTableToSpell(ability)
```

Comment:
shouldn't this be added a few lines above before it is used?

Reply:
The libraries are loaded first, so there is no problem with the order of declaration, however it does not seem intuitive and it makes more sense to have it declared before it is used.

Reply:
Didn't even think about it I'm so use to languages which don't care. Moved the function above the usage.